src/webhook/mutation/pod_mutator.go

```diff
@@ -211,34 +213,44 @@ func (m *podMutator) Handle(ctx context.Context, req admission.Request) admissio
 		return *nsResponse
 	}
 
+	m.currentPodNS = ns.Name
+	defer func() {
+		m.currentPodNS = ""
+	}()
+
 	dk, dkResponse := m.getDynakube(ctx, req, dkName)
 	if dkResponse != nil {
 		return *dkResponse
 	}
 
-	if dk.FeatureDisableMetadataEnrichment() {
+	m.dynaKube = dk
+	defer func() {
+		m.dynaKube = dynatracev1beta1.DynaKube{}
+	}()
```

Comment:
this defer function seems unnecessary

Reply:
@meik99 @chrismuellner I've removed all deferred functions where podMutator fields are nullified.